### PR TITLE
API Tweaks

### PR DESCRIPTION
**Pull request by rodsan0**
Additions to SignalGP and MatchBin APIs.

### CODE REVIEW

File: source/hardware/EventDrivenGP.h [hunk 1]
```diff
@@ -147,7 +147,14 @@ namespace emp {
 
       Event & operator=(const Event &) = default;
       Event & operator=(Event &&) = default;
-
+      bool operator<(const Event & other) const {
+        return std::tie(id, affinity)
+          < std::tie(other.id, other.affinity);
+      }
```
Comment (mmore500) [inline source/hardware/EventDrivenGP.h:153]: I'd mention in the docstring that we're not comparing by one of the elements (the message?)

File: source/tools/MatchBin.h [hunk 2]
```diff
@@ -856,25 +859,27 @@ namespace emp {
     /// Set up regulators to match target MatchBin
     void ImprintRegulators(
```
Comment (mmore500) [inline source/tools/MatchBin.h:860]: Docstring args

File: source/tools/MatchBin.h [hunk 3]
```diff
@@ -856,25 +859,27 @@ namespace emp {
...
       const BaseMatchBin<Val, query_t, tag_t, Regulator> & target
-    ) override {
+    ) override { ImprintRegulators(target.GetState()); }
+
+    void ImprintRegulators(const state_t & target) override {
```
Comment (mmore500) [inline source/tools/MatchBin.h:864]: docstring args

File: tests/hardware/event_driven_gp.cc [hunk 4]
```diff
@@ -172,6 +172,37 @@ TEST_CASE("Test Event Driven GP", "[Hardware]")
 	REQUIRE(ss.str() == "[Message,00000000,(),(Properties: add)]");
 	ss.str(std::string());
 
+  // Event copy assignment operator
+  auto ev2 = ev;
+  gp.TriggerEvent(ev2);
+  gp.PrintEvent(ev2, ss);
+	REQUIRE(ss.str() == "[Message,00000000,(),(Properties: add)]");
```
Comment (mmore500) [inline tests/hardware/event_driven_gp.cc:179]: ?????

File: tests/hardware/event_driven_gp.cc [hunk 5]
```diff
@@ -172,6 +172,37 @@ TEST_CASE("Test Event Driven GP", "[Hardware]")
...
+	ss.str(std::string());
+
+  // Event move assignment operator
+  auto ev3 = std::move(ev2);
+  gp.TriggerEvent(ev3);
+  gp.PrintEvent(ev3, ss);
+	REQUIRE(ss.str() == "[Message,00000000,(),(Properties: add)]");
```
Comment (mmore500) [inline tests/hardware/event_driven_gp.cc:186]: tab vs spaces

File: tests/tools/MatchBin.cc [hunk 6]
```diff
@@ -795,6 +795,41 @@ TEST_CASE("Test MatchBin", "[tools]")
   REQUIRE(bitBin64.GetTags(bitBin64.Match(bs9, 5)) == (emp::vector<emp::BitSet<64>> {bs9, bs1, bs7}));
 
   }
+  // test ImprintRegulators
+  {
+    emp::Random rand(1);
```
Comment (mmore500) [inline tests/tools/MatchBin.cc:800]: comment: setup template matchbin

File: tests/tools/MatchBin.cc [hunk 7]
```diff
@@ -795,6 +795,41 @@ TEST_CASE("Test MatchBin", "[tools]")
...
+    const size_t hi1 = bin1.Put("hi", 1);
+    const size_t bye1 = bin1.Put("bye", 2);
+
+    bin1.SetRegulator(hi1, 0.1);
+    bin1.SetRegulator(bye1, 0.2);
+
+    REQUIRE(bin1.GetRegulator(hi1).state == 0.1);
+    REQUIRE(bin1.GetRegulator(bye1).state == 0.2);
+
+    emp::Random rand2(1);
```
Comment (mmore500) [inline tests/tools/MatchBin.cc:817]: delete rand

Comment (mmore500) [inline tests/tools/MatchBin.cc:817]: comment: set up matchbin to imprint on template

File: tests/tools/MatchBin.cc [hunk 8]
```diff
@@ -795,6 +795,41 @@ TEST_CASE("Test MatchBin", "[tools]")
...
+      std::string,
+      emp::AbsDiffMetric,
+      emp::RouletteSelector<>,
+      emp::LegacyRegulator
+    > bin2(rand2);
+
+    const size_t hi2 = bin2.Put("hi", 1);
+    const size_t bye2 = bin2.Put("bye", 2);
+
+    bin2.ImprintRegulators(bin1);
```
Comment (mmore500) [inline tests/tools/MatchBin.cc:828]: comment: now this matchbin's regulators should match template

File: source/tools/MatchBin.h (hunk 3 above)
Comment (rodsan0) [reply source/tools/MatchBin.h:867]: fixed on 798bdf6

File: tests/tools/MatchBin.cc (hunk 7 above)
Comment (rodsan0) [reply tests/tools/MatchBin.cc]: added on 798bdf6

Comment (rodsan0) [reply tests/tools/MatchBin.cc]: done on 798bdf6

File: tests/tools/MatchBin.cc (hunk 6 above)
Comment (rodsan0) [reply tests/tools/MatchBin.cc:801]: added on 798bdf6

File: tests/tools/MatchBin.cc (hunk 8 above)
Comment (rodsan0) [reply tests/tools/MatchBin.cc:833]: added on 798bdf6

File: tests/hardware/event_driven_gp.cc (hunk 5 above)
Comment (rodsan0) [reply tests/hardware/event_driven_gp.cc:186]: 'fixed' indentation on f0566c6

File: tests/hardware/event_driven_gp.cc (hunk 4 above)
Comment (rodsan0) [reply tests/hardware/event_driven_gp.cc:179]: fixed by f0566c6

File: source/tools/MatchBin.h (hunk 3 above)
Comment (rodsan0) [reply source/tools/MatchBin.h:867]: b41cd2c

File: source/tools/MatchBin.h (hunk 2 above)
Comment (rodsan0) [reply source/tools/MatchBin.h:861]: b41cd2c

File: source/hardware/EventDrivenGP.h (hunk 1 above)
Comment (rodsan0) [reply source/hardware/EventDrivenGP.h:157]: done in 
b41cd2c

**Pull request merged**